### ISSUE TITLE
rstudio-2024.07.0.108 fails to compile "xml-rewrite-3.py: No such file or directory"

### ISSUE THREAD

**Issue by sylvyo** [jgenco/jgenco-overlay#25]
I fails to compile (two weeks ago the same version worked so I think in some changes in gentoo portage).

It exits in this way:

Build type      Release
Install path    /usr
Compiler flags:
C               -O2 -march=znver3 -pipe
C++             -O2 -march=znver3 -pipe -std=c++0x
Linker flags:
Executable      -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs
Module          -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs
Shared          -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs

-- Configuring done (2.3s)
-- Generating done (0.1s)
-- Build files have been written to: /var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6_build
>>> Source configured.
>>> Compiling source in /var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6 ...
Rewriting attributes
/var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/temp/environment: line 2385: /usr/lib64/javatoolkit/bin/xml-rewrite-3.py: No such file or directory
 * ERROR: sci-mathematics/rstudio-2024.07.0.108::jgenco failed (compile phase):
 *   xml-rewrite-3 failed: ./src/gwt/build.xml
 * 
 * Call stack:
 *     ebuild.sh, line  136:  Called src_compile
 *   environment, line 4719:  Called java-pkg-2_src_compile
 *   environment, line 2477:  Called eant 'build' '-f' 'src/gwt/build.xml' '-Dbuild.dir=bin' '-Dwww.dir=www' '-Dextras.dir=extras' '-Dlib.dir=lib' '-Dgwt.main.module=org.rstudio.studio.RStudio' '-DlocalWorkers=14'
 *   environment, line 1256:  Called java-ant-2_src_configure
 *   environment, line 2286:  Called java-ant_bsfix
 *   environment, line 2304:  Called java-ant_bsfix_files './src/gwt/lib/gwt/gwt-rstudio/samples/build.xml' './src/gwt/build.xml'
 *   environment, line 2385:  Called die
 * The specific snippet of code:
 *               ${rewriter3} "${files[@]}" -c --source-element ${JAVA_PKG_BSFIX_SOURCE_TAGS// / --source-element } --source-attribute source --source-value ${want_source} --target-element ${JAVA_PKG_BSFIX_TARGET_TAGS// / --target-element } --target-attribute target --target-value ${want_target} --target-attribute nowarn --target-value yes "${bsfix_extra_args[@]}" || die "xml-rewrite-3 failed: ${file}";
 * 
 * If you need support, post the output of `emerge --info '=sci-mathematics/rstudio-2024.07.0.108::jgenco'`,
 * the complete build log and the output of `emerge -pqv '=sci-mathematics/rstudio-2024.07.0.108::jgenco'`.
!!! When you file a bug report, please include the following information:
GENTOO_VM=openjdk-bin-8  CLASSPATH="" JAVA_HOME="/opt/openjdk-bin-8.412_p08"
JAVACFLAGS="-source 1.8 -target 1.8" COMPILER=""
and of course, the output of emerge --info =rstudio-2024.07.0.108
 * The complete build log is located at '/var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/temp/environment'.
 * Working directory: '/var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6'
 * S: '/var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6'

**Comment by sylvyo** [comment]
emerge --info

Portage 3.0.64 (python 3.12.3-final-0, default/linux/amd64/23.0/split-usr/desktop/plasma, gcc-14, glibc-2.39-r7, 6.9.1-gentoo x86_64)
=================================================================
System uname: Linux-6.9.1-gentoo-x86_64-AMD_Ryzen_9_5900HS_with_Radeon_Graphics-with-glibc2.39
KiB Mem:    32263408 total,  11992844 free
KiB Swap:   34814972 total,  33499388 free
Timestamp of repository steam-overlay: Mon, 20 May 2024 17:23:25 +0000
Head commit of repository steam-overlay: 3aa8d3201d7bce32a1a0fffdf4afd9dbd0583048

Timestamp of repository kzd: Fri, 31 May 2024 17:03:50 +0000
Head commit of repository kzd: d2d4f31f05f0d45abbebc4e35352634ee27e4fad

Timestamp of repository booboo: Sat, 01 Jun 2024 10:06:03 +0000
Head commit of repository booboo: cec0586415f1667881dcfe1c64c37542a0c6ebc7

Head commit of repository jgenco: f6cec6fcd3099ece35e8aec83ffed9f711d11ff8

Timestamp of repository pentoo: Sat, 01 Jun 2024 14:03:50 +0000
Head commit of repository pentoo: c084e87bc4edc837ad772d9212c3d343beae9609

Timestamp of repository gentoo: Sun, 02 Jun 2024 16:30:01 +0000
Head commit of repository gentoo: 10e61f9331cc7671068445fb19e69d669b7570f0
sh bash 5.2_p26-r3
ld GNU ld (Gentoo 2.42 p3) 2.42.0
app-misc/pax-utils:        1.3.7::gentoo
app-shells/bash:           5.2_p26-r3::gentoo
dev-build/autoconf:        2.13-r8::gentoo, 2.72-r1::gentoo
dev-build/automake:        1.16.5-r2::gentoo
dev-build/cmake:           3.29.3::gentoo
dev-build/libtool:         2.4.7-r4::gentoo
dev-build/make:            4.4.1-r1::gentoo
dev-build/meson:           1.4.1::gentoo
dev-java/java-config:      2.3.3-r1::gentoo
dev-lang/perl:             5.38.2-r6::gentoo
dev-lang/python:           3.10.14_p1-r1::gentoo, 3.11.9-r1::gentoo, 3.12.3-r1::gentoo, 3.13.0_beta1_p3::gentoo
dev-lang/rust:             1.78.0::gentoo
sys-apps/baselayout:       2.15::gentoo
sys-apps/openrc:           0.54::gentoo
sys-apps/sandbox:          2.38::gentoo
sys-devel/binutils:        2.42-r1::gentoo
sys-devel/binutils-config: 5.5::gentoo
sys-devel/clang:           14.0.6-r4::gentoo, 17.0.6::gentoo, 18.1.6::gentoo
sys-devel/gcc:             13.3.0::gentoo, 14.1.1_p20240518::gentoo
sys-devel/gcc-config:      2.11::gentoo
sys-devel/lld:             17.0.6::gentoo
sys-devel/llvm:            14.0.6-r4::gentoo, 17.0.6::gentoo, 18.1.6::gentoo
sys-kernel/linux-headers:  6.9::gentoo (virtual/os-headers)
sys-libs/glibc:            2.39-r7::gentoo
Repositories:

steam-overlay
    location: /var/db/repos/steam-overlay
    sync-type: git
    sync-uri: https://github.com/gentoo-mirror/steam-overlay.git
    masters: gentoo
    priority: -1005
    volatile: False

kzd
    location: /var/db/repos/kzd
    sync-type: git
    sync-uri: https://github.com/gentoo-mirror/kzd.git
    masters: gentoo steam-overlay
    priority: -1004
    volatile: False

booboo
    location: /var/db/repos/booboo
    sync-type: git
    sync-uri: https://github.com/gentoo-mirror/booboo.git
    masters: gentoo
    priority: -1003
    volatile: False

jgenco
    location: /var/db/repos/jgenco
    sync-type: git
    sync-uri: https://github.com/jgenco/jgenco-overlay
    masters: gentoo
    priority: -1002
    volatile: False

pentoo
    location: /var/db/repos/pentoo
    sync-type: git
    sync-uri: https://github.com/gentoo-mirror/pentoo.git
    masters: gentoo
    priority: -1001
    volatile: False

gentoo
    location: /var/db/repos/gentoo
    sync-type: rsync
    sync-uri: rsync://rsync.gentoo.org/gentoo-portage
    priority: -1000
    volatile: False
    sync-rsync-verify-metamanifest: yes
    sync-rsync-verify-jobs: 1
    sync-rsync-verify-max-age: 24
    sync-rsync-extra-opts: 

localrepo
    location: /var/db/repos/localrepo
    masters: gentoo
    volatile: False

ACCEPT_KEYWORDS="amd64 ~amd64"
ACCEPT_LICENSE="*"
CBUILD="x86_64-pc-linux-gnu"
CFLAGS="-O2 -march=znver3 -pipe"
CHOST="x86_64-pc-linux-gnu"
CONFIG_PROTECT="/etc /usr/lib64/libreoffice/program/sofficerc /usr/share/config /usr/share/gnupg/qualified.txt"
CONFIG_PROTECT_MASK="/etc/ca-certificates.conf /etc/dconf /etc/env.d /etc/fonts/fonts.conf /etc/gconf /etc/gentoo-release /etc/revdep-rebuild /etc/sandbox.d /etc/terminfo /etc/texmf/language.dat.d /etc/texmf/language.def.d /etc/texmf/updmap.d /etc/texmf/web2c"
CXXFLAGS="-O2 -march=znver3 -pipe"
DISTDIR="/var/cache/distfiles"
ENV_UNSET="CARGO_HOME DBUS_SESSION_BUS_ADDRESS DISPLAY GDK_PIXBUF_MODULE_FILE GOBIN GOPATH PERL5LIB PERL5OPT PERLPREFIX PERL_CORE PERL_MB_OPT PERL_MM_OPT XAUTHORITY XDG_CACHE_HOME XDG_CONFIG_HOME XDG_DATA_HOME XDG_RUNTIME_DIR XDG_STATE_HOME"
FCFLAGS="-O2 -march=znver3 -pipe"
FEATURES="assume-digests binpkg-docompress binpkg-dostrip binpkg-logs buildpkg-live config-protect-if-modified distlocks ebuild-locks fixlafiles ipc-sandbox merge-sync merge-wait multilib-strict network-sandbox news parallel-fetch pid-sandbox pkgdir-index-trusted preserve-libs protect-owned qa-unresolved-soname-deps sandbox sfperms strict unknown-features-warn unmerge-logs unmerge-orphans userfetch userpriv usersandbox usersync xattr"
FFLAGS="-O2 -march=znver3 -pipe"
GENTOO_MIRRORS="https://gentoo.mirror.garr.it/ https://mirror.init7.net/gentoo/"
LANG="it_IT.utf8"
LDFLAGS="-Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs"
LEX="flex"
LINGUAS="it en it_IT en_US"
MAKEOPTS="--jobs 14 --load-average 14"
PKGDIR="/var/cache/binpkgs"
PORTAGE_CONFIGROOT="/"
PORTAGE_RSYNC_OPTS="--recursive --links --safe-links --perms --times --omit-dir-times --compress --force --whole-file --delete --stats --human-readable --timeout=180 --exclude=/distfiles --exclude=/local --exclude=/packages --exclude=/.git"
PORTAGE_TMPDIR="/var/tmp"
SHELL="/bin/bash"
USE="X a52 aac acl acpi activities alsa amd64 bluetooth branding bzip2 cairo cdda cdr cet crypt cups dbus declarative djvu dri dri3 dts dvd dvdr elogind encode exif ffmpeg fftw flac gdbm gif gles2 gpm gtk gui hidpi iconv icu id3tag ipv6 javafx jpeg kde kdesu kwallet lcms libinput libnotify libtirpc mad mng mp3 mp4 mpeg mtp multilib ncurses networkmanager nls ntfs objc ogg opengl openmp opus pam pango pcre pcsc-lite pdf pipewire plasma png policykit ppds pulseaudio qml qt5 readline s3tc samba screencast sdl seccomp semantic-desktop sound spell split-usr ssl startup-notification svg test-rust theora tiff touchpad truetype udev udisks unicode upower usb uxa v4l vaapi vdpau vorbis vpx vulkan wayland widgets wxwidgets x264 xattr xcb xft xml xv xvid xvmc zlib" ABI_X86="64 32" ADA_TARGET="gcc_12" ALSA_CARDS="snd_hda_intel" APACHE2_MODULES="authn_core authz_core socache_shmcb unixd actions alias auth_basic authn_anon authn_dbm authn_file authz_dbm authz_groupfile authz_host authz_owner authz_user autoindex cache cgi cgid dav dav_fs dav_lock deflate dir env expires ext_filter file_cache filter headers include info log_config logio mime mime_magic negotiation rewrite setenvif speling status unique_id userdir usertrack vhost_alias" CALLIGRA_FEATURES="karbon sheets words" CAMERAS="*" COLLECTD_PLUGINS="df interface irq load memory rrdtool swap syslog" CPU_FLAGS_X86="aes avx avx2 f16c fma3 mmx mmxext pclmul popcnt rdrand sha sse sse2 sse3 sse4_1 sse4_2 sse4a ssse3" ELIBC="glibc" GPSD_PROTOCOLS="ashtech aivdm earthmate evermore fv18 garmin garmintxt gpsclock greis isync itrax mtk3301 ntrip navcom oceanserver oncore rtcm104v2 rtcm104v3 sirf skytraq superstar2 tsip tripmate tnt ublox" INPUT_DEVICES="joystick wacom synaptics evdev libinput" KERNEL="linux" L10N="it en" LCD_DEVICES="bayrad cfontz glk hd44780 lb216 lcdm001 mtxorb text" LUA_SINGLE_TARGET="lua5-1" LUA_TARGETS="lua5-1" OFFICE_IMPLEMENTATION="libreoffice" PHP_TARGETS="php8-2" POSTGRES_TARGETS="postgres15" PYTHON_SINGLE_TARGET="python3_12" PYTHON_TARGETS="python3_12" RUBY_TARGETS="ruby31 ruby32" SANE_BACKENDS="*" VIDEO_CARDS="amdgpu radeonsi nvidia vesa v4l v4l2 modesetting fbdev" XTABLES_ADDONS="quota2 psd pknock lscan length2 ipv4options ipp2p iface geoip fuzzy condition tarpit sysrq proto logmark ipmark dhcpmac delude chaos account"
Unset:  ADDR2LINE, AR, ARFLAGS, AS, ASFLAGS, CC, CCLD, CONFIG_SHELL, CPP, CPPFLAGS, CTARGET, CXX, CXXFILT, ELFEDIT, EMERGE_DEFAULT_OPTS, EXTRA_ECONF, F77FLAGS, FC, GCOV, GPROF, INSTALL_MASK, LC_ALL, LD, LFLAGS, LIBTOOL, MAKE, MAKEFLAGS, NM, OBJCOPY, OBJDUMP, PORTAGE_BINHOST, PORTAGE_BUNZIP2_COMMAND, PORTAGE_COMPRESS, PORTAGE_COMPRESS_FLAGS, PORTAGE_RSYNC_EXTRA_OPTS, PYTHONPATH, RANLIB, READELF, RUSTFLAGS, SIZE, STRINGS, STRIP, YACC, YFLAGS

**Comment by jgenco** [comment]
I believe this is a bug in an "old" version of `dev-java/javatoolkit-0.6.7`.  Gentoo Bug [933467](https://bugs.gentoo.org/show_bug.cgi?id=933467#c3) which deals with `media-libs/opencv` has a similar problem and references the now closed Gentoo Bug [933409](https://bugs.gentoo.org/show_bug.cgi?id=933409) which was fixed by  [commit](https://github.com/gentoo/gentoo/commit/80a512317dac9e26be293b7fddb4633fec6d5ff9)

Upgrading to `dev-java/javatoolkit-0.6.7-r1` should fix it.

**Comment by sylvyo** [comment]
Now it goes on but I get another error  (infact in last days I got this error several times, until last time when I got the error I pasted in the first post)

[rstudio_log.txt](https://github.com/user-attachments/files/15596164/rstudio_log.txt)


[65/601] /usr/bin/x86_64-pc-linux-gnu-g++ -DBOOST_ASIO_DISABLE_KQUEUE -DBOOST_ATOMIC_DYN_LINK -DBOOST_ATOMIC_NO_LIB -DBOOST_CHRONO_DYN_LINK -DBOOST_CHRONO_NO_LIB -DBOOST_DATE_TIME_DYN_LINK -DBOOST_DATE_TIME_NO_LIB -DBOOST_ENABLE_ASSERT_HANDLER -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_FILESYSTEM_NO_LIB -DBOOST_IOSTREAMS_DYN_LINK -DBOOST_IOSTREAMS_NO_LIB -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_PROGRAM_OPTIONS_NO_LIB -DBOOST_RANDOM_DYN_LINK -DBOOST_RANDOM_NO_LIB -DBOOST_REGEX_DYN_LINK -DBOOST_REGEX_NO_LIB -DBOOST_SIGNALS_NO_DEPRECATION_WARNING -DBOOST_SYSTEM_DYN_LINK -DBOOST_SYSTEM_NO_LIB -DBOOST_THREAD_DYN_LINK -DBOOST_THREAD_NO_LIB -DFMT_SHARED -DRSTUDIO_BIN_PATH=\"/usr/bin\" -DRSTUDIO_BOOST_NAMESPACE=boost -DRSTUDIO_BOOST_SIGNALS_VERSION=2 -DRSTUDIO_HAS_SOCI -DRSTUDIO_PACKAGE_BUILD -DRSTUDIO_RESOURCES_DIR=\"/usr/share/rstudio\" -DWEBSOCKETPP_STRICT_MASKING -D_FORTIFY_SOURCE=2 -D_WEBSOCKETPP_NO_CPP11_MEMORY_=1 -DSTRIPPED_FILENAME=\"src/cpp/core/http/RequestParser.cpp\" -I/var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6/src/cpp/core/include -I/var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6/src/cpp/shared_core/include -I/var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6/src/cpp/ext -I/var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6_build/src/cpp/core -I/var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6/src/cpp/tests/cpp -isystem /opt/rstudio-tools/x86_64/crashpad -isystem /opt/rstudio-tools/x86_64/crashpad/crashpad -isystem /opt/rstudio-tools/x86_64/crashpad/crashpad/third_party/mini_chromium/mini_chromium -isystem /include  -O2 -march=znver3 -pipe -std=c++0x -std=c++11 -fPIC   -Wall -pthread -Werror=return-type -Wformat -Wformat-security -fstack-protector-strong -fPIC -Wlogical-op -Wduplicated-cond -Wduplicated-branches -Wrestrict -Wnull-dereference -MD -MT src/cpp/core/CMakeFiles/rstudio-core.dir/http/RequestParser.cpp.o -MF src/cpp/core/CMakeFiles/rstudio-core.dir/http/RequestParser.cpp.o.d -o src/cpp/core/CMakeFiles/rstudio-core.dir/http/RequestParser.cpp.o -c /var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6/src/cpp/core/http/RequestParser.cpp
[66/601] /usr/bin/x86_64-pc-linux-gnu-g++ -DBOOST_ASIO_DISABLE_KQUEUE -DBOOST_ATOMIC_DYN_LINK -DBOOST_ATOMIC_NO_LIB -DBOOST_CHRONO_DYN_LINK -DBOOST_CHRONO_NO_LIB -DBOOST_DATE_TIME_DYN_LINK -DBOOST_DATE_TIME_NO_LIB -DBOOST_ENABLE_ASSERT_HANDLER -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_FILESYSTEM_NO_LIB -DBOOST_IOSTREAMS_DYN_LINK -DBOOST_IOSTREAMS_NO_LIB -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_PROGRAM_OPTIONS_NO_LIB -DBOOST_RANDOM_DYN_LINK -DBOOST_RANDOM_NO_LIB -DBOOST_REGEX_DYN_LINK -DBOOST_REGEX_NO_LIB -DBOOST_SIGNALS_NO_DEPRECATION_WARNING -DBOOST_SYSTEM_DYN_LINK -DBOOST_SYSTEM_NO_LIB -DBOOST_THREAD_DYN_LINK -DBOOST_THREAD_NO_LIB -DFMT_SHARED -DRSTUDIO_BIN_PATH=\"/usr/bin\" -DRSTUDIO_BOOST_NAMESPACE=boost -DRSTUDIO_BOOST_SIGNALS_VERSION=2 -DRSTUDIO_HAS_SOCI -DRSTUDIO_PACKAGE_BUILD -DRSTUDIO_RESOURCES_DIR=\"/usr/share/rstudio\" -DWEBSOCKETPP_STRICT_MASKING -D_FORTIFY_SOURCE=2 -D_WEBSOCKETPP_NO_CPP11_MEMORY_=1 -DSTRIPPED_FILENAME=\"src/cpp/core/json/JsonRpc.cpp\" -I/var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6/src/cpp/core/include -I/var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6/src/cpp/shared_core/include -I/var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6/src/cpp/ext -I/var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6_build/src/cpp/core -I/var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6/src/cpp/tests/cpp -isystem /opt/rstudio-tools/x86_64/crashpad -isystem /opt/rstudio-tools/x86_64/crashpad/crashpad -isystem /opt/rstudio-tools/x86_64/crashpad/crashpad/third_party/mini_chromium/mini_chromium -isystem /include  -O2 -march=znver3 -pipe -std=c++0x -std=c++11 -fPIC   -Wall -pthread -Werror=return-type -Wformat -Wformat-security -fstack-protector-strong -fPIC -Wlogical-op -Wduplicated-cond -Wduplicated-branches -Wrestrict -Wnull-dereference -MD -MT src/cpp/core/CMakeFiles/rstudio-core.dir/json/JsonRpc.cpp.o -MF src/cpp/core/CMakeFiles/rstudio-core.dir/json/JsonRpc.cpp.o.d -o src/cpp/core/CMakeFiles/rstudio-core.dir/json/JsonRpc.cpp.o -c /var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6/src/cpp/core/json/JsonRpc.cpp
FAILED: src/cpp/core/CMakeFiles/rstudio-core.dir/json/JsonRpc.cpp.o 
/usr/bin/x86_64-pc-linux-gnu-g++ -DBOOST_ASIO_DISABLE_KQUEUE -DBOOST_ATOMIC_DYN_LINK -DBOOST_ATOMIC_NO_LIB -DBOOST_CHRONO_DYN_LINK -DBOOST_CHRONO_NO_LIB -DBOOST_DATE_TIME_DYN_LINK -DBOOST_DATE_TIME_NO_LIB -DBOOST_ENABLE_ASSERT_HANDLER -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_FILESYSTEM_NO_LIB -DBOOST_IOSTREAMS_DYN_LINK -DBOOST_IOSTREAMS_NO_LIB -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_PROGRAM_OPTIONS_NO_LIB -DBOOST_RANDOM_DYN_LINK -DBOOST_RANDOM_NO_LIB -DBOOST_REGEX_DYN_LINK -DBOOST_REGEX_NO_LIB -DBOOST_SIGNALS_NO_DEPRECATION_WARNING -DBOOST_SYSTEM_DYN_LINK -DBOOST_SYSTEM_NO_LIB -DBOOST_THREAD_DYN_LINK -DBOOST_THREAD_NO_LIB -DFMT_SHARED -DRSTUDIO_BIN_PATH=\"/usr/bin\" -DRSTUDIO_BOOST_NAMESPACE=boost -DRSTUDIO_BOOST_SIGNALS_VERSION=2 -DRSTUDIO_HAS_SOCI -DRSTUDIO_PACKAGE_BUILD -DRSTUDIO_RESOURCES_DIR=\"/usr/share/rstudio\" -DWEBSOCKETPP_STRICT_MASKING -D_FORTIFY_SOURCE=2 -D_WEBSOCKETPP_NO_CPP11_MEMORY_=1 -DSTRIPPED_FILENAME=\"src/cpp/core/json/JsonRpc.cpp\" -I/var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6/src/cpp/core/include -I/var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6/src/cpp/shared_core/include -I/var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6/src/cpp/ext -I/var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6_build/src/cpp/core -I/var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6/src/cpp/tests/cpp -isystem /opt/rstudio-tools/x86_64/crashpad -isystem /opt/rstudio-tools/x86_64/crashpad/crashpad -isystem /opt/rstudio-tools/x86_64/crashpad/crashpad/third_party/mini_chromium/mini_chromium -isystem /include  -O2 -march=znver3 -pipe -std=c++0x -std=c++11 -fPIC   -Wall -pthread -Werror=return-type -Wformat -Wformat-security -fstack-protector-strong -fPIC -Wlogical-op -Wduplicated-cond -Wduplicated-branches -Wrestrict -Wnull-dereference -MD -MT src/cpp/core/CMakeFiles/rstudio-core.dir/json/JsonRpc.cpp.o -MF src/cpp/core/CMakeFiles/rstudio-core.dir/json/JsonRpc.cpp.o.d -o src/cpp/core/CMakeFiles/rstudio-core.dir/json/JsonRpc.cpp.o -c /var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6/src/cpp/core/json/JsonRpc.cpp
/var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6/src/cpp/core/json/JsonRpc.cpp: In member function ‘bool rstudio::core::json::JsonRpcResponse::hasAfterResponse() const’:
/var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6/src/cpp/core/json/JsonRpc.cpp:196:11: error: cannot convert ‘const boost::function<void()>’ to ‘bool’ in return
  196 |    return afterResponse_;
      |           ^~~~~~~~~~~~~~
      |           |
      |           const boost::function<void()>
In file included from /usr/include/boost/function.hpp:16,
                 from /var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6/src/cpp/shared_core/include/shared_core/Logger.hpp:34,
                 from /var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6/src/cpp/shared_core/include/shared_core/Error.hpp:34,
                 from /var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6/src/cpp/shared_core/include/shared_core/json/Json.hpp:36,
                 from /var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6/src/cpp/core/include/core/json/JsonRpc.hpp:24,
                 from /var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-eef4efa0b4a9a6c6d984912a09cd6504decfb8c6/src/cpp/core/json/JsonRpc.cpp:16:
/usr/include/boost/function/function_template.hpp:783:14: note: explicit conversion function was not considered
  783 |     explicit operator bool () const { return !this->empty(); }
      |              ^~~~~~~~

**Comment by sylvyo** [comment]
I think it could be a problem of "boost"

**Comment by jgenco** [comment]
I emerged `dev-libs/boost-1.85.0-r1` and saw the same thing.
Try adding the following patch to the appropriate [patches folder](https://wiki.gentoo.org/wiki//etc/portage/patches).
```patch
--- a/src/cpp/core/json/JsonRpc.cpp
+++ b/src/cpp/core/json/JsonRpc.cpp
@@ -193,7 +193,7 @@ void JsonRpcResponse::setAfterResponse(
    
 bool JsonRpcResponse::hasAfterResponse() const
 {
-   return afterResponse_;
+   return afterResponse_ != nullptr;
 }
    
    

```
It both installed  and started successfully with USE="panmirror qt6 quarto server -clang -debug -doc -electron -qt5 -test"
Note: qt5 doesn't run correctly for me(unrelated to  RStudio), so I now use qt6.  I don't expect any problem for qt5 builds.

I'll need to add this to `sci-mathematics/rstudio-2024.04.0.735` and make sure it works with `~dev-libs/boost-1.84.0` before pushing this.

**Comment by sylvyo** [comment]
I put the patch but when I try to update Manifest I get:

* ERROR: sci-mathematics/rstudio-2024.07.0.108::localrepo failed (depend phase):
 *   npm.eclass could not be found by inherit()
 * 
 * Call stack:
 *                      ebuild.sh, line 632:  Called source '/var/db/repos/localrepo/sci-mathematics/rstudio/rstudio-2024.07.0.108.ebuild'
 *   rstudio-2024.07.0.108.ebuild, line   6:  Called inherit 'cmake' 'llvm' 'java-pkg-2' 'java-ant-2' 'multiprocessing' 'pam' 'qmake-utils' 'xdg-utils' 'npm' 'prefix'
 *                      ebuild.sh, line 277:  Called die
 * The specific snippet of code:
 *   		[[ -z ${location} ]] && die "${1}.eclass could not be found by inherit()"
 * 
 * If you need support, post the output of `emerge --info '=sci-mathematics/rstudio-2024.07.0.108::localrepo'`,
 * the complete build log and the output of `emerge -pqv '=sci-mathematics/rstudio-2024.07.0.108::localrepo'`.
 * The ebuild environment file is located at '/var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/temp/environment'.
 * Working directory: '/var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/empty'
 * S: '/var/tmp/portage/sci-mathematics/rstudio-2024.07.0.108/work/rstudio-2024.07.0.108'

**Comment by sylvyo** [comment]
Sorry, my mistake, it is compiling. I'll update you as soon as it will finish

**Comment by jgenco** [comment]
Good to here.  I've pushed the patch and updated ebuilds.  

The boost patch in your patches folder can now be deleted.

**Comment by jgenco** [comment]
FYI: I have updated the boost patch to copy OpenSUSE's [patch](https://build.opensuse.org/projects/openSUSE:Factory/packages/rstudio/files/fix-boost-1.85-build.patch).  Using Boost's official bool operator seem a lot smarter.
